Library/Homebrew/cmd/update-bash.sh

```diff
@@ -305,7 +323,7 @@ EOS
   wait
   trap - SIGINT
 
-  for DIR in "$HOMEBREW_REPOSITORY" "$HOMEBREW_LIBRARY"/Taps/*/*
+  for DIR in "${DIRS[@]}"
   do
     [[ -d "$DIR/.git" ]] || continue
     pull "$DIR"
```

Comment:
It does more than just `reset` (e.g. cleaning up untracked files). I also don't think it's intuitive that on first run `pull` won't be run at all for the Homebrew repository. This is why I (still) prefer my original version which is a much simpler fix and adds less branching and complexity to the code.